forge/test/mlir/llama/test_llama_inference.py

```diff
@@ -126,3 +127,141 @@ def test_llama_inference_cache_cpu():
     # Generated text
     generated_text = tokenizer.decode(generated_tokens[0], skip_special_tokens=True)
     print(generated_text)
+
+
+@pytest.mark.xfail()
+def test_llama_inference_no_cache_tt():
+
+    # Load Llama 3B model and tokenizer
+    model_path = "openlm-research/open_llama_3b"
+    framework_model = load_model(model_path, use_cache=False)
+    tokenizer = LlamaTokenizer.from_pretrained(model_path)
+
+    # Prepare input sentence
+    prompt = "Q: What is the largest animal?\nA:"
+    inputs = tokenizer(prompt, return_tensors="pt")
+
+    # Run Prefill on CPU with no cache to get the initial logits
+    logits = framework_model(input_ids=inputs.input_ids, attention_mask=inputs.attention_mask)
+    next_token_logits = logits[0][:, -1, :]
+    next_token = torch.argmax(next_token_logits, dim=-1)
+
+    input_ids = torch.cat(
+        [inputs.input_ids, next_token.unsqueeze(0)], dim=-1
+    )
+
+
+    # Run decode stage on TT device and generate tokens by appending predicted token into sequence of input tokens
+    # untill the a specified maximum number of new tokens is reached or an end-of-sequence token is encountered.
+    max_new_tokens = 32
+    for _ in range(max_new_tokens):
+
+        # CPU Inference
+        framework_output = framework_model(input_ids)
+
+        # Compile the model
+        tt_inputs = [input_ids]
+        compiled_model = forge.compile(framework_model, sample_inputs=tt_inputs)
+
+        # Run on TT device
+        tt_output = compiled_model(*tt_inputs)
+        tt_output = [tt_out.to("cpu") for tt_out in tt_output]
+
+        # Validate TT result with Framework
+        assert all([compare_with_golden_pcc(golden=fw_out, calculated=tt_out, pcc=0.99) for fw_out, tt_out in zip(framework_output, tt_output)])
+        next_token_logits = tt_output[0][:, -1, :]
+        next_token = torch.argmax(next_token_logits, dim=-1)
+
+        if next_token == tokenizer.eos_token_id:
+            break
+
+        input_ids = torch.cat(
+            [input_ids, next_token.unsqueeze(0)], dim=-1
+        )
+
+
+    # Generated text
+    generated_text = tokenizer.decode(input_ids[0], skip_special_tokens=True)
+    print("generated_text=",generated_text)
+
+
+@pytest.mark.xfail()
+def test_llama_inference_cache_tt():
+
+    # Load Llama 3B model and tokenizer
+    model_path = "openlm-research/open_llama_3b"
+    framework_model = load_model(model_path, use_cache=True)
+    tokenizer = LlamaTokenizer.from_pretrained(model_path)
+
+    class Llama_Model_Wrapper(torch.nn.Module):
+        def __init__(self, model):
+            super().__init__()
+            self.model = model
+        def forward(self, inputs):
+            past_key_values_list = inputs[1:]
+            past_key_values = list()
+            for idx in range(0, len(past_key_values_list), 2):
```

Comment:
Make sure to explain new PK formating in wrapper description :))

Reply:
Removed the input formating because of unbundling input and past key values as two inputs to the model forward function and added the comment for output formating.